internal/operator-controller/applier/boxcutter.go

```diff
@@ -0,0 +1,291 @@
+				Name: prevRevision.Name,
+				UID:  prevRevision.UID,
+			})
+		}
+
+		if err := controllerutil.SetControllerReference(ext, newRevision, bc.Scheme); err != nil {
+			return false, "", fmt.Errorf("set ownerref: %w", err)
+		}
+		if err := bc.Client.Create(ctx, newRevision); err != nil {
+			return false, "", fmt.Errorf("creating new Revision: %w", err)
+		}
+		currentRevision = newRevision
+	}
+
+	// TODO: Delete archived previous revisions over a certain revision limit
+
+	// TODO: Define constants for the ClusterExtensionRevision condition types.
+	progressingCondition := meta.FindStatusCondition(currentRevision.Status.Conditions, "Progressing")
+	availableCondition := meta.FindStatusCondition(currentRevision.Status.Conditions, "Available")
+	succeededCondition := meta.FindStatusCondition(currentRevision.Status.Conditions, "Succeeded")
+
+	if progressingCondition == nil && availableCondition == nil && succeededCondition == nil {
+		return false, "New revision created", nil
+	} else if progressingCondition != nil && progressingCondition.Status == metav1.ConditionTrue {
+		return false, progressingCondition.Message, nil
+	} else if availableCondition != nil && availableCondition.Status != metav1.ConditionTrue {
+		return false, "", errors.New(availableCondition.Message)
+	} else if succeededCondition != nil && succeededCondition.Status != metav1.ConditionTrue {
+		return false, succeededCondition.Message, nil
+	}
+	return true, "", nil
+}
+
+// getExistingRevisions returns the list of ClusterExtensionRevisions for a ClusterExtension with name extName in revision order (oldest to newest)
+func (bc *Boxcutter) getExistingRevisions(ctx context.Context, extName string) ([]ocv1.ClusterExtensionRevision, error) {
+	existingRevisionList := &ocv1.ClusterExtensionRevisionList{}
+	if err := bc.Client.List(ctx, existingRevisionList, client.MatchingLabels{
+		controllers.ClusterExtensionRevisionOwnerLabel: extName,
+	}); err != nil {
+		return nil, fmt.Errorf("listing revisions: %w", err)
+	}
+	slices.SortFunc(existingRevisionList.Items, func(a, b ocv1.ClusterExtensionRevision) int {
+		return cmp.Compare(a.Spec.Revision, b.Spec.Revision)
+	})
+	return existingRevisionList.Items, nil
+}
+
+// computeSHA256Hash returns a sha236 hash value calculated from object.
+func computeSHA256Hash(obj any) string {
+	hasher := sha256.New()
+	deepHashObject(hasher, obj)
+	return hex.EncodeToString(hasher.Sum(nil))
+}
+
+// deepHashObject writes specified object to hash using the spew library
+// which follows pointers and prints actual values of the nested objects
+// ensuring the hash does not change when a pointer changes.
+func deepHashObject(hasher hash.Hash, objectToWrite any) {
+	hasher.Reset()
+
+	// TODO: change this out to `json.Marshal`. Pretty sure we found issues in the past where
+	//   spew would produce different output when internal structures changed without the
+	//   external public API changing.
```

Comment:
I think this is something we should fix before we merge to `main`. I'll submit a PR to this branch.